### PR TITLE
Fixes for nvmf boot

### PR DESCRIPTION
**Pull request by mwilck**
## Changes

This patch set is a collection of bug fixes for the 95nvmf module. I encountered these issues while working on testing and improving boot from NVMeoF. 

## Checklist
- [x] I have tested it locally
- [x] I have reviewed and updated any documentation if relevant

### CODE REVIEW

**Comment by mwilck** [comment]
Notification for @aafeijoo-suse, @tblume, @hreinecke.

**Comment by aafeijoo-suse** [comment]
CC @lnykryn, ICYI

File: modules.d/95nvmf/parse-nvmf-boot-connections.sh [hunk 1]
```diff
@@ -90,7 +62,9 @@ parse_nvmf_discover() {
         warn "traddr is mandatory for $trtype"
         return 0
     fi
-    if [ "$trtype" = "fc" ]; then
+    if [ "$trtype" = "tcp" ]; then
+        : > /tmp/nvmf_needs_network
+    elif [ "$trtype" = "fc" ]; then
         if [ "$traddr" = "auto" ]; then
```
[the remark this answers is not in the document]
Comment (mwilck) [reply modules.d/95nvmf/parse-nvmf-boot-connections.sh:68]: > Is "auto" valid only for "fc"?

Yes

>     * If yes, `man/dracut.cmdline.7.asc` should be updated accordingly.

I'll see what I can do.

File: modules.d/95nvmf/module-setup.sh [hunk 2]
```diff
@@ -76,9 +81,25 @@ cmdline() {
...
+            [[ -z $trtype || $trtype != "${trtype#NQN}" ]] && continue
+            unset traddr
+            unset host_traddr
+            unset trsvcid
+            if [[ $_address =~ .*,.* ]]; then
+                # nvme-cli 2.0: comma-separated
+                mapfile -t -d ',' _address_parts < <(printf "%s" "$_address")
+            else
+                # nvme-cli 1.0: space-separated
+                _address_parts=("$_address" "$_a2" "$_a3")
```
[the remark this answers is not in the document]
Comment (mwilck) [reply modules.d/95nvmf/module-setup.sh:94]: Well we can't assume every tool is always at it's latest release, can we? Given that it's rather simple for dracut to support both versions, I see no reason why we shouldn't.

[the remark this answers is not in the document]
Comment (johannbg) [reply modules.d/95nvmf/module-setup.sh:94]: > Well we can't assume every tool is always at it's latest release, can we?
Yes we can.
Any downstream distribution that thinks it's smart and wants to play russian release roulette by mixing and matching various releases of the components that make up the core/baseOS is on their own and have to carry/support/maintain that shenanigan downstream meanwhile we follow other upstream releases closely as we should ( kernel,systemd,util-linux, etc ).

[the remark this answers is not in the document]
Comment (mwilck) [reply modules.d/95nvmf/module-setup.sh:94]: Fair enough, no problem to remove this part if you guys dislike it. I would still like to find out who, or what, defines the combination of "components that make up the core/baseOS". Is is always the latest release of every component? Or is it what's currently in Fedora Rawhide, or some other distribution?

[the remark this answers is not in the document]
Comment (mwilck) [reply modules.d/95nvmf/module-setup.sh]: I dropped the commit "fix(nvmf): support nvme-cli 1.x list-subsys output" for now.

**Comment by mwilck** [comment]
Closing.

**Pull request closed without merging**